integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DataMapJobs.scala

```diff
@@ -64,9 +79,13 @@ class DistributedDataMapJob extends AbstractDataMapJob {
       filterInf = removeSparkUnknown(filterInf,
         dataMapFormat.getCarbonTable.getAbsoluteTableIdentifier, filterProcessor)
       dataMapFormat.setFilterResolverIntf(filterInf)
-      IndexServer.getClient.getSplits(dataMapFormat).toList.asJava
+      IndexServer.getClient.getSplits(dataMapFormat)
+        .getExtendedBlockets(dataMapFormat.getCarbonTable.getTablePath, dataMapFormat.getQueryId)
     }
     LOGGER.info(s"Time taken to get response from server: $time ms")
+    if (isTempFolderCreated && !file.delete()) {
```

Comment:
Please move to finally to delete the temp folder

Reply:
fixed